docs/pages/CodeAnalysis.md

```diff
@@ -0,0 +1,226 @@
+- sap.ui.require.preload (restricted)
+
+The [JSModule Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/JSModuleAnalyzer.js) uses the ["Abstract Syntax Tree"](https://en.wikipedia.org/wiki/Abstract_syntax_tree) (AST) of a JavaScript file to decide whether a code block is executed *conditionally* or *unconditionally*.
+
+The analyzer uses a set of rules to decide whether one of the above APIs is immediately called when the module is executed or whether the API is only called under certain conditions.
+
+For example, top-level code is always executed. Flow-control statements in JavaScript imply that certain blocks of code are only executed under certain conditions (e.g. if block, else block, ...). Besides those inherent JavaScript rules, further common patterns are known to the analyzer, e.g. immediately invoked function expressions or the factor function of AMD modules.
+
+Any dependencies that are found in code that - according to those rules - is always executed, are collected as eager (or standard) dependencies. Dependencies that are found on a code path that depends on certain conditions are collected as conditional dependencies.
+
+The bundling implemented by the UI5 tooling can either follow only eager dependencies (resolve:`true`) or also conditional dependencies (resolveConditional). For further information see [Custom Bundling](https://sap.github.io/ui5-tooling/v3/pages/Configuration/#custom-bundling).
+
+When a dependency in one of the mentioned APIs is not a "Simple Literal" but an expression, the corresponding module is marked as "having dynamic dependencies". This marker is currently not further evaluated by the tooling.
+
+### Component Analyzer
+
+The [Component Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/ComponentAnalyzer.js) analyzes JavaScript files named `Component.js` to collect dependency information by searching for a `manifest.json` in the same folder. If one is found, the `sap.ui5` section will be evaluated in the following way
+
+- any library dependency is added as a dependency to the `library.js` module of that library. If the library dependency is modeled as 'lazy', the
+ module dependency will be added as 'conditional'
+- any component dependency is added as a dependency to a file named `Component.js`. If the dependency is modeled as 'lazy', the module dependency will be added as 'conditional'
+- for each configured UI5 model, for which a type is configured, a module dependency to that type is added
+- for each route that contains a view name, a module dependency to that view will be added
+
+### Fiori Elements Analyzer
+
+The [Fiori Elements Analyzer](https://github.com/SAP/ui5-builder/blob/main/lib/lbt/analyzer/FioriElementsAnalyzer.js) analyzes a FioriElements app and its underlying template components to collect dependency information. It searches for a `manifest.json` in the same folder. If one is found the `sap.fe` section will be evaluated in the following way:
```

Comment:
"It searches for a `manifest.json` in the same folder."

Same folder as what?

Reply:
@KlattG What about "It searches for a `manifest.json` located in the same folder as the `Component.js`"

Reply:
That sounds fine, thanks!

Reply:
Done